util/httputil/compression_test.go

```diff
@@ -138,6 +143,98 @@ func TestCompressionHandler_Deflate(t *testing.T) {
 	require.NoError(t, err, "unexpected error while reading the response body")
 
 	actual := buf.String()
-	expected := "Hello World!"
-	require.Equal(t, expected, actual, "expected response with content")
+	require.Equal(t, respBody, actual, "expected response with content")
+}
+
+func Benchmark_compression(b *testing.B) {
```

Comment:
We have `BenchmarkGzip` elsewhere; does this add enough to justify 90 extra lines?

Reply:
I wanted to have more an "integration benchmark" as `BenchmarkGzip` call the `gzip.NewReader/gzip.NewWriter` directly instead of calling the `http.Handler` (note that there is optimizations also inside the [GzipHandler](https://github.com/klauspost/compress/blob/9a951bee0e5e9df261e6ca6a68bfa1e819f333d0/gzhttp/compress.go#L420) like pooling the writers.

WDYT?